src/Stream.re

```diff
@@ -383,9 +384,9 @@ module Readable = {
                   (
                     t('r),
                     Js.nullable(Js.Exn.t),
-                    (~err: option(Js.Exn.t)) => unit
+                    (~err: option(Js.Exn.t)) => calledBackDestroy
                   ) =>
-                  unit
+                  calledBackDestroy
```

Comment:
Where would the user create this type `calledBackDestroy`? I ask because this function takes a user-defined function, and so the user needs to have some way to create this type inside the callback they implement and pass to the API.

Reply:
You're right! - I was too quick to apply the same pattern to everything. A constructor for such type needs to be provided, which admittedly yields bad DX. If a uniform api-surface is a goal, I'm not sure how to proceed. I will apply this pattern only to callbacks provided by node (and not the dev) to see how it looks & feels.

Reply:
So, I rechecked in more detail and I'm not sure I see your issue: `Stream_test.makePlainReadable` still compiles without an issue on this (pr-)branch.

Reply:
@woeps, Sorry for the delayed reply. I'm currently in the middle of moving to a new city, so I've been swamped with other concerns lately. I am currently reviewing the PR. I will get back to you on this ASAP. Thanks for being patient.

Reply:
@woeps, I see what you mean. The `destroy` event passes an anonymous callback to the handler. The callback is not implemented by the user, but by Node's internal code, so the user doesn't need to create this type. I think it works well for this particular use case.

I just wrote some tests which exercise the `Readable.destroy` functionality, including the error case, and it seems to work properly.

I will work on merging this in the coming days. Thanks for the PR!